travis/travis_weblate.py

```diff
@@ -1,166 +1,253 @@
-    addons_list = list(main_depends)
-    database = "openerp_test"
-    # Use by default version 10 connection context
-    connection_context = context_mapping.get(odoo_version, Odoo10Context)
-    current_branch = travis_branch
-    wlprojects = get_projects(travis_repo_shortname, current_branch)
-    try:
-        # first project found
-        wlproject = wlprojects.next()
-    except StopIteration as sie:
-        print(red("Project Weblate %s not found for branch %s" % (
-            travis_repo_shortname, current_branch)))
-        raise sie
-    with connection_context(server_path, addons_path, database) \
-            as odoo_context, lock(wlproject, addons_list):
-        try:
-            subprocess.check_output([
-                'git', 'remote', 'add', 'travis',
-                'https://%(GH_TOKEN)s@github.com/%(REPO_SLUG)s' % dict(
-                    GH_TOKEN=gh_token, REPO_SLUG=travis_repo_slug)])
-        except subprocess.CalledProcessError:
-            # Remote created previously
-            pass
-        subprocess.check_output(['git', 'config', 'user.name', "Travis CI"])
-        subprocess.check_output([
-            'git', 'config', 'user.email', "moylop260@vauxoo.com"])
-        if wl_push(wlproject):
-            command = ['git', 'pull', 'travis', current_branch]
-            res = subprocess.check_output(command).strip('\n ')
-            print(yellow("git pull result: %s" % res))
-
-        command = ['git', 'log', '-r', '-1', '--oneline']
-        sha = subprocess.check_output(command).strip('\n ')
-        print(yellow("Current sha %s" % (sha)))
-
-        for module in addons_list:
+from test_server import (get_test_dependencies as get_depends, get_addons_path,
+                         get_addons_to_check, get_server_path, parse_list)
+from apis import ApiException, WeblateApi, GitHubApi
+from travis_helpers import yellow
+from git_run import GitRun
+
+
+class TravisWeblateUpdate(object):
+
+    GIT_COMMIT_INFO = {
+        'author': 'Weblate bot <weblate@bot>',
+        'message': '[REF] i18n: Updating translation terms from weblate '
+                   '[ci skip]'
+    }
+
+    def __init__(self):
+        self._git = GitRun(os.path.join(os.getcwd(), '.git'), True)
+        self.branch = os.environ.get("TRAVIS_BRANCH",
+                                     self._git.get_branch_name())
+        remote = self._git.run(["ls-remote", "--get-url", "origin"])
+        remote = remote.replace(':', '/')
+        if '@' in remote:
+            remote = re.sub('.+@', '', remote)
+        remote = re.sub('.git$', '', remote)
+        match_object = re.search(
+            r'(?P<host>[^/]+)/(?P<owner>[^/]+)/(?P<repo>[^/]+)', remote)
+        if match_object:
+            host = match_object.group("host").replace(
+                'https://', '').replace('http://', '')
+            owner = match_object.group("owner")
+            repo = match_object.group("repo")
+            remote = '%(host)s:%(owner)s-%(repo)s' % {
```

Comment:
@moylop260 Done 
The name of the project is `github.com:ORG/project(branch)` and the slug is `github_com_OCA_project_branch_`